server/src/main/java/org/elasticsearch/index/mapper/TimeSeriesIdFieldMapper.java

```diff
@@ -82,7 +82,22 @@ public TimeSeriesIdFieldMapper build() {
         }
     }
 
-    public static final TypeParser PARSER = new ConfigurableTypeParser(c -> new TimeSeriesIdFieldMapper(), c -> new Builder());
+    /**
+     * A no-op builder does not return an instance of the field mapper. Instead, it returns null
+     * This is going to be used for non-time-series indices that should have this
+     *
+     */
+    public static class NoopBuilder extends Builder {
+        @Override
+        public TimeSeriesIdFieldMapper build() {
+            return null;
+        }
+    }
+
+    public static final TypeParser PARSER = new ConfigurableTypeParser(
+        c -> c.getIndexSettings().getMode() == IndexMode.TIME_SERIES ? new TimeSeriesIdFieldMapper() : null,
+        c -> c.getIndexSettings().getMode() == IndexMode.TIME_SERIES ? new Builder() : new NoopBuilder()
+    );
```

Comment:
I think you can use a FixedTypeParser here if there's no configuration

Reply:
I've been trying to avoid tests for `getMode() == IndexMode.TIME_SERIES` so that when you read `IndexMode` you can see all of the behaviors that it changes. I'm not sure the right way to do that here, but I'd sure like to.